src/controllers/api/inventoryController.ts

```diff
@@ -1,11 +1,28 @@
-import inventory from "@/static/fixed_responses/inventory.json";
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import { toInventoryResponse } from "@/src/helpers/inventoryHelpers";
+import { Inventory } from "@/src/models/inventoryModel";
 import { Request, RequestHandler, Response } from "express";
 
-const inventoryController: RequestHandler = (request: Request, response: Response) => {
-    console.log(request.query);
+const inventoryController: RequestHandler = async (request: Request, response: Response) => {
     const accountId = request.query.accountId;
+
+    if (!accountId) {
+        response.status(400).json({ error: "accountId was not provided" });
+        return;
+    }
     console.log(accountId);
-    response.json(inventory);
+
+    const inventory = await Inventory.findOne({ accountOwnerId: accountId }); // has the accountOwnerId field to find a corresponding inventory
+    const inventoryJSON = inventory?.toJSON();
+
+    if (!inventoryJSON) {
+        response.status(400).json({ error: "inventory was undefined" });
+        return;
+    }
+
+    const inventoreResponse = toInventoryResponse(inventoryJSON); // remove the accountOwnerId for the response
```

Comment:
check if inventory is undefined before trying `toJson()`

Reply:
?. checks for undefined!

Reply:
that's why you check before so you dont two undefined checks

Reply:
good suggestion!

Reply:
I understand now what you mean